File: CodingChallengeSolutionsC++/781. Rabbits in Forest/781.RabbitsinForest.js
/**
 * @param {number[]} answers
 * @return {number}
 */
var numRabbits = function (answers) {
    unique = new Map();
    rabbits = 0;

    for (x = 0; x < answers.length; ++x) {
        if (!unique.has(answers[x])) {
            unique.set(answers[x], 1);
        }
        else {
            unique.set(answers[x], unique.get(answers[x]) + 1);
        }
    }

    unique.forEach(function (value, key) {
        rabbits += Math.ceil(value / (key + 1)) * (key + 1)
    })

    return rabbits;
};

File: CodingChallengeSolutionsC++/781. Rabbits in Forest/781.RabbitsinForest.cs


namespace CodingChallengeSolutions.RabbitsinForest
{
    public class RabbitsinForest
    {
        public int NumRabbits(int[] answers)
        {
            Dictionary<int, int> unique = new Dictionary<int, int>();
            int rabbits = 0;
            for (int x = 0; x < answers.Count(); ++x)
            {
                if (!unique.ContainsKey(answers[x]))
                {
                    unique.Add(answers[x], 1);
                }
                else
                {
                    unique[answers[x]]++;
                }

            }

            foreach (KeyValuePair<int, int> entry in unique)
            {
                rabbits += (int)Math.Ceiling((double)entry.Value / (entry.Key + 1)) * (entry.Key + 1);
            }
            return rabbits;
        }
    }
}
